staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/aes/aes.go

```diff
@@ -43,44 +53,135 @@ import (
 // therefore transformers using this implementation *must* ensure they allow for frequent key
 // rotation. Future work should include investigation of AES-GCM-SIV as an alternative to
 // random nonces.
-type gcm struct {
-	block cipher.Block
-}
+func NewGCMTransformer(block cipher.Block) (value.Transformer, error) {
+	aead, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
 
-// NewGCMTransformer takes the given block cipher and performs encryption and decryption on the given
-// data.
-func NewGCMTransformer(block cipher.Block) value.Transformer {
```

Comment:
do we have a test that captures data encrypted by a previously released version of this and ensures the current version can still decrypt it?

Reply:
I will add an explicit one.  Note that I didn't not touch decryption so it should be fine since existing tests are passing.

Reply:
> Note that I didn't not touch decryption

I know that was the intent... enough stuff moved around that I'm having trouble eyeballing it and being 100% certain

Reply:
> > do we have a test that captures data encrypted by a previously released version of this and ensures the current version can still decrypt it?
> 
> I didn't see a fixture of obviously encrypted data... did I miss this test?

[TestGCMLegacyDataCompatibility](https://github.com/kubernetes/kubernetes/pull/116155/files#diff-e8f886d4a6f00f2bcc161c87a21332c574f1092958f8be274caf6e2287cad95dR189-R190)

https://github.com/kubernetes/kubernetes/blob/02031085033a88f49a190ce45c79459edb5adca4/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/aes/aes_test.go#L189-L190

> > Note that I didn't not touch decryption
> 
> I feel like that is a good representation of my brain after reviewing this PR

😂 😭

Reply:
> // recorded output from NewGCMTransformer at commit 3b1fc60d8010dd8b53e97ba80e4710dbb430beee

thanks, verified that package hasn't changed since v1.26.0, so it's representative of released state